target/mips/gdbstub.c

```diff
@@ -194,9 +195,9 @@ int mips_gdb_get_cheri_reg(CPUMIPSState *env, GByteArray *mem_buf, int n)
         return gdb_get_capreg(mem_buf, &env->active_tc.CHWR.EPCC);
     case 41:
         return gdb_get_capreg(mem_buf, &env->active_tc.CHWR.ErrorEPCC);
-    case CHERI_GDB_NUM_CAPREGS + 1:
+    case CHERI_GDB_NUM_CAPREGS:
         return gdb_get_regl(mem_buf, env->CP2_CapCause);
-    case CHERI_GDB_NUM_CAPREGS + 2: {
+    case CHERI_GDB_NUM_CAPREGS + 1: {
```

Comment:
I suspect that only this change is what is actually needed to fix `cap_valid` without any of the changes to GDB itself.  I don't understand though why this would only have broken now though.

Reply:
Looks like this changed from hardcoded 42/43 to these values in some of the RISC-V changes in February.

I just tried making only this change and it did indeed fix QEMU for me with stock CHERI gdb:

```
mips_wait ()
    at /usr/home/john/work/cheri/git/cheribsd/sys/mips/mips/exception.S:775
775             jr      ra
(gdb) info registers 
                  zero               at               v0               v1
 R0   0000000000000000 0000000000000000 0000000000000000 fffffffffffffff8 
                    a0               a1               a2               a3
 R4   ffffffff809d0280 0000000000000000 ffffffff808a4ccd 0000000000000312 
                    a4               a5               a6               a7
 R8   0000000000000000 0000000000000000 0000000000000100 0000000000000000 
                    t0               t1               t2               t3
 R12  00000000400084a0 00000000400084a1 0000000000000000 0000000000000064 
                    s0               s1               s2               s3
 R16  ffffffff80990000 ffffffff80992a58 ffffffff809d0280 ffffffff809d0278 
                    s4               s5               s6               s7
 R20  9800000000b1f024 0000000000000000 0000000000000000 0000000000000040 
                    t8               t9               k0               k1
 R24  0000000000000001 ffffffff804ecf28 0000000000000000 9800000000a26710 
                    gp               sp               s8               ra
 R28  0000000000000000 c00000000e767930 c00000000e767950 ffffffff807d1d44 
                status               lo               hi         badvaddr
      00000000400084a1 000000007d70a3a8 0000000000000000 c000000000ee3ef0 
                 cause               pc
      0000000000000008 ffffffff807ceb00 
                  fcsr              fir
              00000000         00038100 
  c0: 0x0000000000000000
  c1: 0x0000000000000000
  c2: 0x0000000000000000
  c3: 0x0000000000000000
  c4: 0xffffffff8085053f [rwRW,0xffffffff8085053f-0xffffffff80850543]
  c5: 0x0000000000000000
  c6: 0x0000000000000000
  c7: 0x0000000000000000
  c8: 0x0000000000000000
  c9: 0x0000000000000000
 c10: 0x0000000000000000
 c11: 0x0000000000000000
 c12: 0x0000000000000000
 c13: 0x0000000000000000
 c14: 0x0000000000000000
 c15: 0x0000000000000000
 c16: 0x0000000000000000
 c17: 0x0000000000000000
 c18: 0x0000000000000000
 c19: 0x0000000000000000
 c20: 0x0000000000000000
 c21: 0x0000000000000000
 c22: 0x0000000000000000
 c23: 0x0000000000000000
 c24: 0x0000000000000000
 c25: 0x0000000000000000
 c26: 0x0000000000000000
 c27: 0x0000000000000000
 c28: 0x0000000000000000
 c29: 0x0000000000000000
 c30: 0x0000000000000000
 c31: 0x0000000000000000
 ddc: 0x0000000000000000 [rwRW,0x0-0xffffffffffffffff]
 pcc: 0x0000000000000000
             cap_cause        cap_valid             prid
      0000000000000000 0000000100000011 00000000000f0405 
UserTls: 0x0000000000000000
PrivTls: 0x0000000000000000
kr1c: 0x0000000000000000
kr2c: 0x0000000000000000
 kcc: 0x0 [rxR,0x0-0xffffffffffffffff]
 kdc: 0x0000000000000000 [rwRW,0x0-0xffffffffffffffff]
epcc: 0xffffffff807e016c <cpu_switch+804> [rxR,0x0-0xffffffffffffffff]
errorepcc: 0xffffffffbfc00000 [rwxRW,0x0-0xffffffffffffffff]
```

Reply:
I only noticed the off by one after the XML changes. I can probably revert those.